### DISCUSSION TITLE
Why is doing a factory reset not recommended when toltec is installed?

### DISCUSSION THREAD

**Discussion by Groxx** [toltec-dev/toltec#529] [Support] [answered]
The FAQ on toltec-dev.org mentions:

>**Can I factory reset my reMarkable if I have toltec installed?**
> No, DO NOT factory reset your reMarkable if you have toltec installed. First uninstall toltec with `toltecctl uninstall`.

I don't see any mention of _why_ though.  
**What happens if you do end up doing a factory reset?**  And how do you recover from it?  

I'm broadly assuming that it means you'll have to do something like ssh in and re-enable xochitl, so it's just more troublesome than desired, but details would be interesting and potentially useful for anyone who missed that FAQ entry.

**Comment by LinusCDE** [comment, marked as answer]
I think @matteodelabre might know this better, but my assumption is the following:

A factory reset only wipes user data. If this wipes/resets `/home`, this could make your current system unbootable since you might have an alternate launcher selected which now no longer exists (launchers usually have a fall back though). But there are other things that can prevent a boot apart from an alterantive (now missing launcher).  
Reason is that the extra software has "a separate filetree/root" where all libs/binaries are installed to: `/opt`. This is in fact at `/home/root/.entware`. So a binary found at `/opt/bin/somebinary` is in fact at `/home/root/.entware/bin/somebinary`. Removing this directory would not remove the service file that wants to mount that directory to `/opt` and thus could prevent booting.

**Comment by Eeems** [comment]
As @LinusCDE mentioned there are things installed into root that will not be cleaned up when a factory reset is done. This requires users to know what to manually clean up in order to get xochitl started again, and to get back to a factory state. Which is obviously not fair to expect of most users.

**Comment by matteodelabre** [comment]
The only thing I’d add is that users can always do a “real” factory reset by following instructions from <https://github.com/ddvk/remarkable-update>, and get back to a clean system. But this may not be an easy option for all users, which is why we recommend people don’t use the standard factory reset option.
